appgate/syncer/git.py

```diff
@@ -1,108 +1,126 @@
-            shutil.rmtree(self.repository_dir)
-
-        log.info(f"[git-sync] Initializing the git repository by cloning {url}")
-        self.repository = Repo.clone_from(repo_url, self.repository_dir)
-
-    def checkout_branch(self) -> None:
-        self.branch = f'{str(datetime.date.today())}.{time.strftime("%H-%M-%S")}'
+    repository_name: str = attrib()
+    repository_path: Path = attrib()
+    git_repo: Repo = attrib()
+    base_branch: str = attrib()
+    vendor: str = attrib()
+
+    def checkout_branch(self, branch: str, dry_run: bool) -> None:
         log.info(
-            f"[git-syncer] Checking out new branch {self.repository.remote().name}/{self.branch}"
+            f"[git-operator] Checking out new branch {self.git_repo.remote().name}/{branch}"
         )
-        self.repository.git.branch(self.branch)
-        self.repository.git.checkout(self.branch)
+        if dry_run:
+            return
+        self.git_repo.git.branch(branch)
+        self.git_repo.git.checkout(branch)
 
     def needs_pull_request(self) -> bool:
-        self.repository.index.add([f"{self.repository_dir}/*"])
-        return self.repository.is_dirty()
+        self.git_repo.index.add([f"{self.repository_path}/*"])
+        return self.git_repo.is_dirty()
 
-    def commit_change(self) -> None:
+    def commit_change(self, branch: str, dry_run) -> None:
         log.info(
-            f"[git-syncer] Committing changes to {self.repository.remote().name}:{self.branch}"
+            f"[git-operator] Committing changes to {self.git_repo.remote().name}:{branch}"
         )
-        self.repository.index.commit(self.branch)
+        if not dry_run:
+            self.git_repo.index.commit(branch)
 
-    def push_change(self) -> None:
+    def push_change(self, branch: str, dry_run: bool) -> None:
         log.info(
-            f"[git-syncer] Pushing changes to {self.repository.remote().name}:{self.branch}"
-        )
-        self.repository.git.push(
-            "--set-upstream", self.repository.remote().name, self.branch
+            f"[git-operator] Pushing changes to {self.git_repo.remote().name}:{branch}"
         )
+        if not dry_run:
+            self.git_repo.git.push(
+                "--set-upstream", self.git_repo.remote().name, branch
+            )
 
-    def create_pull_request(self) -> None:
+    def create_pull_request(self, branch: str, dry_run: bool) -> None:
         pass
 
 
+def github_repo(ctx: GitOperatorContext, repository_path: Path) -> GitRepo:
+    token = ensure_env(GITHUB_TOKEN_ENV)
+    # Fine-grained token? make sure the user is oauth2
+    if token.startswith("github_pat_"):
+        username = "oauth2"
+    elif not ctx.git_username:
+        raise AppgateException("Unable to find github username.")
+    else:
+        username = ctx.git_username
+    repository = f"github.com:{ctx.git_repository}"
+    log.info(f"[git-operator] Initializing the git repository by cloning {repository}")
```

Comment:
Nitpick but better to keep using %-formatting for logs.
```suggestion
    log.info("[git-operator] Initializing the git repository by cloning %s", repository)
```